packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx

```diff
@@ -258,10 +259,13 @@ export class AtomicSearchBox {
   }
 
   private async triggerSuggestions() {
-    await Promise.all(
-      this.suggestions.map((suggestion) => suggestion.onInput())
+    // eslint-disable-next-line node/no-unsupported-features/es-builtins
+    const settled = await Promise.allSettled(
+      this.suggestions.map((suggestion) => promiseTimeout(suggestion.onInput()))
```

Comment:
I think I would prefer not to implicitly add a time out, especially if they time out silently, since it may be difficult to debug why some particular component works *sometimes* but not always.

However, I would keep the `Promise.allSettled` and I would try to log whenever a promise rejected.

Reply:
so just let it take as long as it takes? I wonder if there should be a loading state, as currently it will just pop up out of nowhere when it's ready, and if it takes like 2s, it's quite a while

Reply:
In this case I think we should prioritize end user experience over dev experience.

There should be some limit to how long we wait, so that end user can benefit from "some" suggestion if one of the provider does not return properly.

When one provider fails to do its job, we should definitely be very verbose in the console as to why something was not shown so that a dev can at least quickly understand what's happening.

This timeout should definitely be configurable (an option on the search box), and have a sane default of ... couple hundred ms... ?